node/state_api/src/lib.rs

```diff
@@ -0,0 +1,397 @@
+pub fn state_reply<DB>(
+    state_manager: &StateManager<DB>,
+    key: &TipsetKeys,
+    cid: &Cid,
+) -> Result<InvocResult<UnsignedMessage>, BoxError>
+where
+    DB: BlockStore,
+{
+    let tipset = ChainStore::new(state_manager.get_block_store()).tipset_from_keys(key)?;
+    let (msg, ret) = call::state_replay(&state_manager, &tipset, cid)?;
+
+    Ok(InvocResult {
+        msg,
+        msg_rct: ret.msg_receipt().clone(),
+        actor_error: ret.act_error().map(|e| e.to_string()),
+    })
+}
+
+/// returns a state tree given a tipset
+pub fn state_for_ts<DB>(
+    state_manager: &StateManager<DB>,
+    maybe_tipset: Option<Tipset>,
+) -> Result<StateTree<DB>, BoxError>
+where
+    DB: BlockStore,
+{
+    let block_store = state_manager.get_block_store_ref();
+    let tipset = if maybe_tipset.is_none() {
+        chain::get_heaviest_tipset(block_store)?
+    } else {
+        maybe_tipset
+    };
+
+    let (st, _) = task::block_on(state_manager.tipset_state(&tipset.unwrap()))?;
+    let state_tree = StateTree::new_from_root(block_store, &st)?;
+    Ok(state_tree)
+}
+
+/// returns the indicated actor's nonce and balance.
+pub fn state_get_actor<DB>(
+    state_manager: &StateManager<DB>,
+    actor: &Address,
+    key: &TipsetKeys,
+) -> Result<Option<actor::ActorState>, BoxError>
+where
+    DB: BlockStore,
+{
+    let tipset = ChainStore::new(state_manager.get_block_store()).tipset_from_keys(key)?;
+    let state = state_for_ts(state_manager, Some(tipset))?;
+    state.get_actor(actor).map_err(|e| e.into())
+}
+
+/// returns the public key address of the given ID address
+pub fn state_account_key<DB>(
+    state_manager: &StateManager<DB>,
+    actor: &Address,
+    key: &TipsetKeys,
+) -> Result<Address, BoxError>
+where
+    DB: BlockStore,
+{
+    let tipset = ChainStore::new(state_manager.get_block_store()).tipset_from_keys(key)?;
+    let state = state_for_ts(state_manager, Some(tipset))?;
+    let address =
+        interpreter::resolve_to_key_addr(&state, state_manager.get_block_store_ref(), actor)?;
+    Ok(address)
+}
+
+/// retrieves the ID address of the given address
+pub fn state_lookup_id<DB>(
+    state_manager: &StateManager<DB>,
+    address: &Address,
+    key: &TipsetKeys,
+) -> Result<Address, BoxError>
+where
+    DB: BlockStore,
+{
+    let tipset = ChainStore::new(state_manager.get_block_store()).tipset_from_keys(key)?;
+    let state = state_for_ts(state_manager, Some(tipset))?;
+    state.lookup_id(address).map_err(|e| e.into())
+}
+
+/// returns the addresses of every actor in the state
+pub fn state_list_actors<DB>(
+    state_manager: &StateManager<DB>,
+    key: &TipsetKeys,
+) -> Result<Vec<Address>, BoxError>
+where
+    DB: BlockStore,
+{
+    let tipset = ChainStore::new(state_manager.get_block_store()).tipset_from_keys(key)?;
+    state_manager::utils::list_miner_actors(&state_manager, &tipset).map_err(|e| e.into())
```

Comment:
shouldn't this list all actors?

Reply:
This function shouldn't be here, been removed.